backend/molgenis-emx2-io/src/main/java/org/molgenis/emx2/io/ImportSchemaEmx1Task.java

```diff
@@ -33,26 +41,66 @@ public static void uploadFromStoreToSchema(TableStore store, Schema targetSchema
     loadRefRelationships(emx1schema, entities, attributes);
 
     // load into target schema
-    targetSchema.migrate(emx1schema);
+    schema.migrate(stripOldName(emx1schema));
+
+    // update message
+    schemaTask.complete();
 
     // revert map
     Map<String, String> tableToSheet = new LinkedHashMap<>();
     for (Map.Entry<String, Emx1Entity> entry : entities.entrySet()) {
       tableToSheet.put(entry.getValue().getName(), entry.getKey());
     }
 
-    // load the tables
+    // load the tables, using the original names
     for (TableMetadata table : emx1schema.getTables()) {
-      if (store.containsTable(tableToSheet.get(table.getTableName()))) {
-        targetSchema
-            .getTable(table.getTableName())
-            .save(store.readTable(tableToSheet.get(table.getTableName()))); // actually upsert
+      String emx1EntityName = tableToSheet.get(table.getTableName());
+      if (store.containsTable(emx1EntityName)) {
+        Task tableTask = this.addSubTask("Loading " + table.getTableName()).start();
+
+        // map for mapping from 'name' to 'label'
+        Map<String, String> attributeMap =
```

Comment:
important, we now use the 'label' instead of the 'id' of the emx1 attribute as name of emx2 column.

This requires data to also be reheadered.